### PR TITLE
IRGen: adjust for LLVM SVN r368000

### PR DESCRIPTION
**Pull request by compnerd**
Replace this paragraph with a description of your changes and rationale. Provide links to external references/discussions if appropriate.

Resolves SR-NNNN.

### CODE REVIEW

**Comment by compnerd** [comment]
CC: @rjmccall

**Comment by compnerd** [comment]
@swift-ci please smoke test

File: lib/IRGen/GenDecl.cpp [hunk 1]
```diff
@@ -4163,7 +4163,8 @@ Address IRGenFunction::createAlloca(llvm::Type *type,
                                     const llvm::Twine &name) {
   llvm::AllocaInst *alloca =
       new llvm::AllocaInst(type, IGM.DataLayout.getAllocaAddrSpace(), ArraySize,
-                           alignment.getValue(), name, AllocaIP);
+                           llvm::MaybeAlign(alignment.getValue()), name,
+                           AllocaIP);
```
Comment (rjmccall) [inline lib/IRGen/GenDecl.cpp]: Do we ever actually use `Alignment(0)` as input to this?

Comment (compnerd) [reply lib/IRGen/GenDecl.cpp]: I'm not certain, but I do agree with you that we shouldn't be passing 0 - the signature for `AllocaInst` has changed and now requires the `MaybeAlign` type (as a wrapper to the alignment).

Comment (rjmccall) [reply lib/IRGen/GenDecl.cpp]: Right, I'm just wondering if maybe we can pass `llvm::Align(alignment.getValue())` (or even have an implicit conversion, and maybe someday unify the types).

**Comment by compnerd** [comment]
@swift-ci please test

**Comment by compnerd** [comment]
@swift-ci please test

**Comment by compnerd** [comment]
@swift-ci please test

**Pull request closed without merging**